### ISSUE TITLE
PMO finalizer blocks the successful deletion of short living CAPI clusters (used in CI)

### ISSUE THREAD

**Issue by bavarianbidi** [giantswarm/roadmap#2296]
### Description

When CI is creating a CAPI-Cluster, PMO add the finalizer `operatorkit.giantswarm.io/prometheus-meta-operator-cluster-api-controller`.
As the cluster has a very short lifecycle it got deleted within a few minutes again.

During the first reconciliation it seems:
* PMO noticed and add finalizer
* Cluster get deleted before PMO has created all resources - _guessing here: the remote write secret_
* now PMO never manage to clean up things
* Cluster CR is stuck

### TODO

- [ ] Add the x to the y to get the z.
- [ ] Update relevant docs.

**Comment by primeroz** [comment]
Cluster currently stuck on glippy: **t-w5saxoa**

created and deleted in 4 minutes
```
➜ k get cluster t-w5saxoa -o yaml | yq '.metadata| [.creationTimestamp,.deletionTimestamp]'
- "2023-04-05T09:32:14Z"
- "2023-04-05T09:36:59Z"
```

**PMO Logs for it**
`logs query --from="2023-04-05T09:00:00Z" --to="2023-04-05T10:00:00Z" --limit=3000 '{installation="glippy",  app="prometheus-meta-operator"} |= "t-w5saxoa" | json message="message" | line_format "{{.message}}"' --no-labels --forward`

[pmo-logs.txt](https://github.com/giantswarm/roadmap/files/11167619/pmo-logs.txt)

**Comment by bdehri** [comment]
There were two clusters stuck at deleting at grizzly with PMO finalizer, I cleaned them up but Pmo still  exits after 30 seconds or so.
[prometheus-meta-op.txt](https://github.com/giantswarm/roadmap/files/11178262/prometheus-meta-op.txt)

**Comment by TheoBrigitte** [comment]
This sounds like a use case where monitoring is not needed and should be disabled https://docs.giantswarm.io/getting-started/observability/monitoring/disable/

**Comment by primeroz** [comment]
I tested https://github.com/giantswarm/prometheus-meta-operator/pull/1255 on glippy and worked fine

* at the time i switched the image to this branch we had a cluster stuck for over 2 hours because of the pmo finalizer, it got cleared and the cluster finally got deleted
* I created a cluster and deleted it as soon as the finalizer was added, this would get the cluster stuck on 4.33 but with this branch the finalizer got cleared as expected

good work, can't wait for it to land :+1: :pray:

**Comment by QuentinBisson** [comment]
Released in 4.35.1
